### DISCUSSION TITLE
Scheduled workflow can't access secrets

### DISCUSSION THREAD

**Discussion by douglatornell** [community/community#25955] [Actions]
<p>I recently added an <code>on: schedule:</code> trigger to a workflow that has been working fine via its <code>on:push:</code> trigger for some time. The workflow sends notifications to a couple of Slack channels via webhook URL stored in secrets. The schedule-triggered action fails to notify the Slack channels with the message <code>##[error]Cannot read property 'url' of undefined</code>.</p>
<p>Here is an example of the failure:<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/SalishSeaCast/NEMO-Cmd/runs/840224581?check_suite_focus=true" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/76797/181094297-14364e1c-902b-4773-b7f0-9ae58a804ff0.png" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/SalishSeaCast/NEMO-Cmd/runs/840224581?check_suite_focus=true" target="_blank" rel="nofollow noopener">SalishSeaCast/NEMO-Cmd</a></h3>

<p>NEMO command processor, a command line tool for doing various operations associated with the running NEMO (and other) ocean models.  This an extensible tool built on the OpenStack cliff (Command Li...</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>This looks to me like the scheduled workflow is being blocked from accessing secrets, but I can’t find anything in the docs that tells me that should be the case. I’m seeing this in several repos, some of which use organization secrets, others that use repo secrets.</p>
<p>Is this a known issue that I am failing to find in my searching, or am I doing something wrong?</p>

**Comment by cschleiden** [comment, marked as answer]
<p><a class="mention" href="/douglatornell">@douglatornell</a> Scheduled workflows do receive secrets. Looking at your example, you can also see the masked secrets (the <code>***</code>), so <em>some value</em> was passed to the action.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/b5f40fcf360aa911c14cbb5b55c757e14ff8f1f0.png" data-download-href="/uploads/short-url/pXDjXXKPEazTzVec3XNZ4ofGDOo.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2201819/181094298-5592a172-e245-4b72-95ce-752f32eb3184.png" alt="image" data-base62-sha1="pXDjXXKPEazTzVec3XNZ4ofGDOo" width="690" height="254" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/b5f40fcf360aa911c14cbb5b55c757e14ff8f1f0_2_690x254.png, https://aws1.discourse-cdn.com/github/original/2X/b/b5f40fcf360aa911c14cbb5b55c757e14ff8f1f0.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/b/b5f40fcf360aa911c14cbb5b55c757e14ff8f1f0.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/b5f40fcf360aa911c14cbb5b55c757e14ff8f1f0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">703×259 55.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The problem is with the action you’re calling, looking at the code it tries to read <code>head_commit</code> from the JSON payload. That doesn’t exist for scheduled workflows and leads to this error:</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/pioug/le-slack-message/blob/087c7c6d7af33952a5b9bb49a72ed90489277cd2/index.js#L20" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/pioug/le-slack-message/blob/087c7c6d7af33952a5b9bb49a72ed90489277cd2/index.js#L20" target="_blank" rel="nofollow noopener">pioug/le-slack-message/blob/087c7c6d7af33952a5b9bb49a72ed90489277cd2/index.js#L20</a></h4>
<pre class="onebox"><code class="lang-js"><ol class="start lines" start="10" style="counter-reset: li-counter 9 ;">
<li>const ACTION_NAME = core.getInput("ACTION_NAME");</li>
<li>const JOB = JSON.parse(core.getInput("JOB", { required: true }));</li>
<li>const SLACK_WEBHOOK_URL = core.getInput("SLACK_WEBHOOK_URL", {</li>
<li>  required: true</li>
<li>});</li>
<li>
</li>
<li>const message = new IncomingWebhook(SLACK_WEBHOOK_URL);</li>
<li>await message.send({</li>
<li>  attachments: [</li>
<li>    {</li>
<li class="selected">      title: `${ACTION_NAME} in &lt;${GITHUB_EVENT.head_commit.url}/checks|${GITHUB_WORKFLOW}&gt;: ${JOB.status}`,</li>
<li>      color: {</li>
<li>        Success: "good",</li>
<li>        Cancelled: "warning",</li>
<li>        Failure: "danger"</li>
<li>      }[JOB.status],</li>
<li>      fields: [</li>
<li>        {</li>
<li>          value: `${GITHUB_EVENT.head_commit.message.split("\n")[0]} _(&lt;${</li>
<li>            GITHUB_EVENT.head_commit.url</li>
<li>          }|${GITHUB_EVENT.ref.replace(</li>
</ol></code></pre>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by douglatornell** [comment]
<p>Thanks <a class="mention" href="/cschleiden">@cschleiden</a> ! I had looked at the action code but didn’t connect the dots about it trying to read a non-existent <code>head_commit</code>. I’ll see if I can engage with the action author to collaborate on a we of resolving this.</p>

**Comment by douglatornell** [comment]
<p>For anyone who find this thread in the future: I couldn’t find an obvious good way to enable <a href="https://github.com/pioug/le-slack-message" rel="nofollow noopener">pioug/le-slack-message@v1.0.0</a> to work for <code>schedule</code> events, so I changed to use <a href="https://github.com/8398a7/action-slack" rel="nofollow noopener">8398a7/action-slack@v3</a> for Slack notifications.</p>
